Framework/Core/src/DataInputDirector.cxx

```diff
@@ -68,8 +73,11 @@ std::regex DataInputDescriptor::getFilenamesRegex()
 
 void DataInputDescriptor::addFileNameHolder(FileNameHolder* fn)
 {
-  if (!mAlienSupport && fn->fileName.rfind("alien://", 0) == 0) {
-    LOG(debug) << "AliEn file requested. Enabling support.";
+  // remove leading file:// from file name
+  if (fn->fileName.rfind("file://", 0) == 0) {
```

Comment:
I guess only this is really needed to work around this bug? Given that `TFile::Open` can open a file with a `file://` prefix just fine, what is in our machinery that breaks it?

Reply:
The crash seems to happen, when the tree retrieved from file://.. is used to initialize the TTreeReader with TTreeReader treeReader{tree} in TableTreeHelpers.cxx. This does not happen without file://. Removing the leading file:// is a fast fix. But I will try to pin down the issue more precisely.

Reply:
what I see is that the file is processed twice if you have file:// in front. Could that be the reason? I agree we should have the fast fix ASAP, could you open a PR which simply drops the prefix?